pkg/api/apis/operators/subscription_types.go

```diff
@@ -35,6 +37,56 @@ type SubscriptionSpec struct {
 	InstallPlanApproval    Approval
 }
 
+// SubscriptionConditionType indicates an explicit state condition about a Subscription in "abnormal-true"
+// polarity form (see https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties).
+type SubscriptionConditionType string
+
+const (
+	// SubscriptionCatalogSourcesUnhealthy indicates that some or all of the CatalogSources to be used in resolution are unhealthy.
+	SubscriptionCatalogSourcesUnhealthy SubscriptionConditionType = "CatalogSourcesUnhealthy"
+)
+
+const (
+	// NoCatalogSourcesFound is a reason string for Subscriptions with unhealthy CatalogSources due to none being available.
+	NoCatalogSourcesFound string = "NoCatalogSourcesFound"
+
+	// AllCatalogSourcesHealthy is a reason string for Subscriptions whose CatalogSources are all healthy.
+	AllCatalogSourcesHealthy string = "AllCatalogSourcesHealthy"
+
+	// CatalogSourceUpdated is a reason string for Subscriptions that transitioned due to a CatalogSource being updated.
+	CatalogSourceUpdated string = "CatalogSourceUpdated"
+
+	// CatalogSourceRemoved is a reason string for Subscriptions that transitioned due to a CatalogSource being removed.
+	CatalogSourceRemoved string = "CatalogSourceRemoved"
+
+	// UnhealthyCatalogSourceFound is a reason string for Subscriptions that transitioned because an unhealthy catalog source was found.
+	UnhealthyCatalogSourceFound string = "UnhealthyCatalogSourceFound"
```

Comment:
I haven't made it this far yet, but ideally we use this in conjunction with a message that indicates whether that matters for the subscription (i.e. it's explicitly referenced in the Subscription spec)